acp/identity/identity.go

```diff
@@ -18,18 +18,31 @@ import (
 	"github.com/sourcenetwork/immutable"
 )
 
+// Identity is the unique identifier for an actor.
+type Identity immutable.Option[string]
```

Comment:
todo: I dislike the introduction of this type for the following reasons:

1) This seems like an independent change to this PR, I don't think Identity Context propagation stuff needs this (correct me if I am wrong).

2) Identity being optional is not and should not be part of the Identity type.

3) `immutable.Option[string]` is more readable IMO. 

4) Introduction of wrapper `HasValue` and `Value` functions which immutable provides already.

5) Currently the `Identity` is just a string, and should stay like that. In future when authentication is implemented, the `Identity` type will evolve into a `struct` type which will then be used like `immutable.Option[Identity]`

6) The abstraction should likely be done when Identity type becomes more complex (likely when authentication is done).

I also noticed that testing frameworks Identity params changed as well, I share similar thoughts for them, they should just be a string for now.

Reply:
The usage of the `Identity` type is to avoid having to change func param types when we change from string to some other type. That being said, we could achieve the same thing and keep the immutability out of the type definition if we did the following instead:
```go
type Identity string
```
end then use `immutable.Option[Identity]` as param type.

Reply:
I agree the `immutable.Option` is the wrong type here. I would like to keep the custom type as it will make future changes simpler. I will look into changing the alias to `type Identity string`.

Reply:
> The usage of the `Identity` type is to avoid having to change func param types when we change from string to some other type. That being said, we could achieve the same thing and keep the immutability out of the type definition if we did the following instead:
> ```go
> type Identity string
> ```
> end then use `immutable.Option[Identity]` as param type.

I am okay with this alias and param type suggestion.

Reply:
Seems like resolved thread but chiming in anyways ;). I agree with Fred/Keenan on this. Using a custom type for this future proofs the changes we know we need to make. But also, yes the type should just be `string` and not `Option[string]`. Cheers